lib/php/lib/Transport/TSocket.php

```diff
@@ -328,7 +328,8 @@ public function write($buf)
             if ($writable > 0) {
                 // write buffer to stream
                 $written = fwrite($this->handle_, $buf);
-                if ($written === -1 || $written === false) {
+                $closed_socket = $written === 0 && feof($this->handle_);
+                if ($written === -1 || $written === false || $closed_socket) {
                     throw new TTransportException(
```

Comment:
$written === false

Not your code, but I always wonder if a language has no boolean operators when I see code like this.